idea/src/org/jetbrains/kotlin/idea/intentions/AddJvmStaticIntention.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.intentions
+
+import com.intellij.codeInsight.intention.LowPriorityAction
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.idea.inspections.UnusedSymbolInspection
+import org.jetbrains.kotlin.idea.util.addAnnotation
+import org.jetbrains.kotlin.idea.util.findAnnotation
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtObjectDeclaration
+
+class AddJvmStaticIntention : SelfTargetingIntention<KtNamedFunction>(
+    KtNamedFunction::class.java,
+    "Add '@JvmStatic' annotation"
+), LowPriorityAction {
+
+    private val annotationFqName = FqName("kotlin.jvm.JvmStatic")
+
+    override fun isApplicableTo(element: KtNamedFunction, caretOffset: Int): Boolean {
+        if (element.findAnnotation(annotationFqName) != null) return false
+        if (!UnusedSymbolInspection.isEntryPoint(element)) return false
```

Comment:
See `KotlinRunLineMarkerContributor` for the correct way to check that a function is `main`.